### ISSUE TITLE
Conditionals and non-existent files: final value is never evaluated.

### ISSUE THREAD

**Issue by mbologna** [hashicorp/terraform#12194]
### Terraform Version
Terraform v0.8.7

### Affected Resource(s)
Core issue.

### Terraform Configuration Files
```hcl
output "test" {
  value = "${ true ? "foo" : file("/tmp/non-existent-file")}"
}
```

### Debug Output
https://gist.github.com/mbologna/85c41f93bf52b11a122f12cf4c6f4e87

### Panic Output
None

### Expected Behavior

```
Outputs:
test = foo
```

### Actual Behavior
No output. Should have been `test = foo`

### Steps to Reproduce
`terraform apply`

### Important Factoids
It seems that both conditionals are evaluated before choosing the final value. This could be a problem.

**Comment by apparentlymart** [comment]
Hi @mbologna! Sorry this isn't working as well as it ought to.

This behavior is, unfortunately, a known limitation of the initial implementation of the conditional syntax: it always evaluates both of its expressions before discarding one. This was an artifact of some pre-existing design in the interpolation language evaluator that would've required more risky/extensive code surgery than we could do in time for the 0.8 release.

The root cause here is the same as for #11566. I've filed terraform/hil#50 to track the root issue here; the "hil" codebase is where the interpolation language infrastructure lives.

**Comment by moio** [comment]
Thanks @apparentlymart! Are there plans to fix hashicorp/hil#50 in Terraform 0.9?

**Comment by mitchellh** [comment]
Yep closing dup of #11566 I know we have other issues to track this. Sorry!
